src/lib.rs

```diff
@@ -505,9 +505,10 @@ entry! {
     #[serde(serialize_with = "serialize_one_or_many_opt")]
     #[serde(deserialize_with = "deserialize_one_or_many_opt")]
     "affiliated" => affiliated: Vec<PersonsWithRoles> | [PersonsWithRoles],
-    /// Publisher of the item.
-    "publisher" => publisher: FormatString,
-    /// Physical location at which the item was published or created.
+    /// Publisher of the item. May be a string or a dictionary with `name` and
+    /// `location` (both strings).
+    "publisher" => publisher: Publisher,
```

Comment:
Not sure if this is the appropriate location to describe the type, maybe we can just have something like "Publisher of the item, which may have a name and a location."